sound/soc/sof/intel/hda-bus.c

```diff
@@ -72,7 +78,14 @@ void sof_hda_bus_init(struct snd_sof_dev *sdev, struct device *dev)
 
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_LINK)
 #if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA_AUDIO_CODEC)
-	snd_hdac_ext_bus_init(bus, dev, &bus_core_ops, sof_hda_ext_ops);
+	const struct sof_intel_dsp_desc *chip = get_chip_info(sdev->pdata);
+
+	if (chip && chip->hw_ip_version < SOF_INTEL_ACE_2_0) {
+		snd_hdac_ext_bus_init(bus, dev, &bus_core_ops, sof_hda_ext_ops);
+	} else {
+		snd_hdac_ext_bus_init(bus, dev, &bus_core_pio_ops, sof_hda_ext_ops);
+		bus->use_pio_for_commands = true;
+	}
```

Comment:
Don't you need a LunarLake-specific quirk for snd-hda-intel as well? CORB/PIO are not SOF-specific.

Reply:
I would actually add a new bit in AZX capabilities and do everything in sound/hda. There's no reason to do anything in SOF regarding commands.

e.g. add something in hda_controller.h 
````
#define  AZX_DCAPS_PIO_COMMANDS 31
````
and add the quirk in the PCI data.

Reply:
> Don't you need a LunarLake-specific quirk for snd-hda-intel as well? CORB/PIO are not SOF-specific.

Yes, we need a separate quirk for legacy HDA, or a way to use PIO command in case of Lunar Lake.
This PR is not complete, it is for SOF so we can run complete validation test on it.

Reply:
> I would actually add a new bit in AZX capabilities and do everything in sound/hda. There's no reason to do anything in SOF regarding commands.

The two stack is independent in this regards, afaik.
We need separate ways to set different `hdac_bus_ops` via legacy and SOF stack _and_ to set the new `use_pio_for_commands` flag at the same time.

> e.g. add something in hda_controller.h
> 
> ```
> #define  AZX_DCAPS_PIO_COMMANDS 31
> ```

The legacy stack do not support this type of operation (hda_controller w/ azx_*), it can fallback to pure PIO mode w/o RIRB and thus no URs, so the legacy is a bit more involved.

> 
> and add the quirk in the PCI data.

Sure, but the legacy (using hda_controller) and SOF (using hdac_controller) do not share code that we could do this in one place... afaik, imho.